src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -424,6 +424,25 @@ public void InitializeService(ServiceHost serviceHost)
                     OwnerUri = executeParams.OwnerUri,
                     BatchSummaries = q.BatchSummaries
                 };
+
+                // If there were no batch results in the query, send back a message that the commands were completed successfully
+                if (eventParams.BatchSummaries.Length == 0)
+                {
+                    eventParams.BatchSummaries = new BatchSummary[] { new BatchSummary()
+                    {
+                        Messages = new ResultMessage[] { new ResultMessage(SR.QueryServiceCompletedSuccessfully) }
+                    }};
+
+                    var batchParams = new QueryExecuteBatchCompleteParams();
+                    batchParams.BatchSummary = eventParams.BatchSummaries[0];
+                    batchParams.OwnerUri = executeParams.OwnerUri;
+
+                    // HACK: this is here to prevent the QueryExecuteCompleteEvent from being sent before the QueryExecuteResult is processed
+                    System.Threading.Thread.Sleep(100);
```

Comment:
Following up offline chat: I'd suggest using a semaphore here. The other event will lift it, this will block on it. That way if already called we don't wait. If it hasn't gone through yet, this will cleanly wait on that.

---
In reply to: [89245031](https://github.com/Microsoft/sqltoolsservice/pull/157#discussion_r89245031) [](ancestors = 89245031)